test/real_content/test_swupd.bats

```diff
@@ -0,0 +1,211 @@
+	fi
+	return
+}
+
+verify_system() {
+	#TODO: FIX
+	# There are validation problems happening in diverse files. Keeping this disabled until we
+	# have this fixed in verify fix.
+
+	#run sudo sh -c "$SWUPD verify --picky $SWUPD_OPTS 2>/dev/null"
+	#if [ ! -z "$output" ]; then
+	#	echo "Verify found extra files in the system:"
+	#	echo "$output"
+	#	return 1
+	#fi
+
+	#run sudo sh -c "$SWUPD verify $SWUPD_OPTS"
+	#assert_not_in_output "did not match"
+
+	return
+}
+
+# check_version: Check if version in /usr/lib/os-release is equal $1.
+# Fails otherwise.
+check_version() {
+	version="$1"
+	cur_version=$(grep VERSION_ID ${ROOT_DIR}/usr/lib/os-release|cut -d = -f 2)
+	if [ "$version" -ne "$cur_version" ]; then
+		echo "Version $version is different from expected $cur_version"
+		return 1
+	fi
+}
+
+@test "test bundle-list --all" {
+	i=$((${NUM_VERSIONS}-1))
+
+	sudo mkdir -p "${ROOT_DIR}/usr/lib"
+	echo "VERSION_ID=${VERSION[$i]}" | sudo tee "${ROOT_DIR}/usr/lib/os-release"
+
+	num_pkgs=$(sudo sh -c "$SWUPD bundle-list --all $SWUPD_OPTS -F $FORMAT | wc -l")
+	num_pkgs_mom=$(sh -c "curl ${URL}/${VERSION[i]}/Manifest.MoM 2>/dev/null | grep ^M\\\. | wc -l")
+
+	if [ "$num_pkgs" -ne "$num_pkgs_mom" ]; then
+		echo "Number of packages on bundle-list --all is $num_pkgs. Expected is $num_pkgs_mom"
+		return 1
+	fi
+}
+
+# Test update in delta pack range
+@test "incremental updates" {
+	echo "Install minimal system with oldest version (${VERSION[0]})"
+	run sudo sh -c "$SWUPD verify --install $SWUPD_OPTS -m ${VERSION[0]} -F $FORMAT"
+	assert_status_is 0
+	check_version ${VERSION[0]}
+
+	echo "Install one package"
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS os-core-update -F $FORMAT"
+	assert_status_is 0
+	verify_system
+
+	echo "Install all packages"
+	all_packages=$(sudo $SWUPD bundle-list --all $SWUPD_OPTS)
+	all_packages=$(echo $all_packages) #take \n out
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS $all_packages"
+	verify_system
+
+	i=1
+	while [ ! -z ${VERSION[$i]} ]; do
+		echo "Update system to next version (${VERSION[$i]})"
+		run sudo sh -c "$SWUPD update $SWUPD_OPTS -m ${VERSION[$i]}"
+		assert_status_is 0
+		check_version ${VERSION[$i]}
+		verify_system
+
+		i=$(($i+1))
+	done
+}
+
+# Test update out of the delta pack range
+@test "update from first to last" {
+	echo "Install minimal system with oldest version (${VERSION[0]})"
+	run sudo sh -c "$SWUPD verify --install $SWUPD_OPTS -m ${VERSION[0]} -F $FORMAT"
+	assert_status_is 0
+	check_version ${VERSION[0]}
+
+	echo "Install one package"
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS os-core-update -F $FORMAT"
+	assert_status_is 0
+	verify_system
+
+	echo "Install all packages"
+	all_packages=$(sudo $SWUPD bundle-list --all $SWUPD_OPTS)
+	all_packages=$(echo $all_packages) #take \n out
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS $all_packages"
+	verify_system
+
+	i=1
+	version=""
+	while [ ! -z ${VERSION[$i]} ]; do
+		version="${VERSION[$i]}"
+		i=$(($i+1))
+	done
```

Comment:
To get the last item in the array, you could do one of the following instead of using a loop:

${VERSION[-1]} # requires bash > v4.1
${VERSION[${#VERSION[@]} -1]}

https://unix.stackexchange.com/questions/198787/is-there-a-way-of-reading-the-last-element-of-an-array-with-bash

Reply:
hahaha. I should have googled that.